File: examples/python/client-example.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import requests
import argparse
import pyotp

from loremipsum import get_sentences

BOB_CLIENT_VERSION = "1.0"
BOB_CLIENT_NAME = "Bob Client Example"

apps = {
    "dev": "https://3-dot-api-dot-dazzling-rex-760.appspot.com/_ah/api/bob/v3",
}

universities = {
    "upn": "Universiad Privada del Norte",
}

parser = argparse.ArgumentParser()

parser.add_argument(
    "app",
    type=str,
    help="Available apps are ({0})".format(", ".join(apps.keys())),
    choices=apps.keys())

parser.add_argument(
    "university",
    type=str,
    help="Available universities are ({0})".format(
        ", ".join(universities.keys())),
    choices=universities.keys())

parser.add_argument(
    "account",
    type=str,
    help="User's e-mail, the account identifier")

args = parser.parse_args()


def default():
    print "%s v%s" % (BOB_CLIENT_NAME, BOB_CLIENT_VERSION)


def create_account(email, university="upn"):
    headers = {
        "Content-Type": "application/json",
        "Accept": "application/json",
        "BobUniversity": "upn",
        "BobToken": ""}

    t = pyotp.TOTP("R3A7PZLCUQIJFUGX", interval=90)
    token = t.now()

    headers["BobToken"] = str(token)

    firstname = get_sentences(1)[0].split(" ")[0]
    paternallastname = get_sentences(1)[0].split(" ")[0]
    maternallastname = get_sentences(1)[0].split(" ")[0]
    password = get_sentences(1)[0].split(" ")[0]
    phonenumber = get_sentences(1)[0].split(" ")[0].upper()

    values = {
        "EmailAddress": email,
        "FirstName": firstname,
        "PaternalLastName": paternallastname,
        "MaternalLastName": maternallastname,
        "Password": password,
        "PhoneNumber": phonenumber,
        "Agreement": True,
        "Origin": university,
        "PortfolioID": ""
    }

    url = "{0}/bob/createaccount".format(apps[args.app])

    print("Creating account for '{0}' from '{1}'...".format(email, university))

    response = requests.post(url, params=values, headers=headers)

    try:
        print("done.")
        print("Json response:")

        print(response.json())
    except:
        print("Not a JSON response")
        print("Failed.")


if __name__ == "__main__":
    default()

    create_account(
        args.account, args.university if args.university else None)

File: examples/php/client-example.php
<?php

    $url = "https://1-dot-dazzling-rex-760.appspot.com/_ah/api/bob/v1/bob/createaccount";

    $bobuniversity = "orienta";
    $bobkey = "094e3089-91e0-4449-b2d1-1110038ad88c";

    $data = json_encode(array(
       'Agreement'             => 'true',
       'EmailAddress'          => $_POST["email"],
       'FirstName'             => $_POST["first_name"],
       'PaternalLastName'      => $_POST["paternal_lastname"],
       'MaternalLastName'      => $_POST["maternal_lastname"],
       'GradeNumber'           => '5',
       'Origin'                => $bobuniversity,
       'Password'              => $_POST["password"],
       'PhoneNumber'           => $_POST["phone"],
       'UserName'              => $_POST["email"]
    ));

    $headers = array(
      'Accept: application/json',
      'Content-Type: application/json',
      'BobUniversity: $bobuniversity',
      "BobKey: $bobkey"
    );

    var_dump($headers);
    var_dump($data);

    $resp = CallAPI("POST", $url, $data, $headers);

    var_dump($resp);

    function CallAPI($method, $url, $data, $headers) {
        $curl = curl_init();

        curl_setopt($curl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
        curl_setopt($curl, CURLOPT_POST, true);
        curl_setopt($curl, CURLOPT_URL, $url);
        curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
        curl_setopt($curl, CURLOPT_POSTFIELDS, $data);
        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
        curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);

        $result = curl_exec($curl);

        curl_close($curl);

        return $result;
    }

?>
